apps/design/backend/migrations/1758581797749_add-quick-results-table.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * @type {import('node-pg-migrate').ColumnDefinitions | undefined}
+ */
+exports.shorthands = undefined;
+
+/**
+ * @param pgm {import('node-pg-migrate').MigrationBuilder}
+ * @param run {() => void | undefined}
+ * @returns {Promise<void> | void}
+ */
+exports.up = (pgm) => {
+  // Create a table for quick results reporting to store reported results.
+  pgm.createTable(
+    'results_reports',
+    {
+      machine_id: { type: 'text', notNull: true },
+      election_id: { type: 'text', notNull: true },
+      ballot_hash: { type: 'text', notNull: true },
+      is_live_mode: { type: 'boolean', notNull: true },
+      signed_at: {
+        type: 'timestamptz',
+        notNull: true,
+      },
+      encoded_compressed_tally: { type: 'text', notNull: true },
+      precinct_id: { type: 'text' }, // Null for non-precinct-specific results
+    },
+    {
+      constraints: {
+        primaryKey: ['ballot_hash', 'machine_id', 'is_live_mode'],
```

Comment:
Do we want this to be per ballot hash or per election ID? Do we allow multiple elections with the same ballot hash?

Reply:
I went with ballot hash since thats how we will look up the data. My understanding that I checked w/ Jonah is that whenever an election is cloned/duplicated a new ID will generate, since the election id is part of what is in the election json blob that gets hashed that should guarantee unique ballot hashes across elections. And it should guarantee that if two results have the same ballot_hash they also have the same election_id.